flink-connectors/flink-connector-kafka-0.11/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducer011.java

```diff
@@ -957,6 +928,11 @@ private void recycleTransactionalProducer(FlinkKafkaProducer<byte[], byte[]> pro
 		return initProducer(registerMetrics);
 	}
 
+	private FlinkKafkaProducer<byte[], byte[]> initNonTransactionalProducer(boolean registerMetrics) {
+		producerConfig.remove("transactional.id");
```

Comment:
why do you need this change?

Reply:
Because otherwise the the restore of state from the `EXACTLY_ONCE` case will insert this key in the producer config and we will end up having a producer that is configured for transactions, which we don't want.